### ISSUE TITLE
python:3.7 image contains python3.9 under /usr/bin

### ISSUE THREAD

**Issue by mmraz** [docker-library/python#744]
After pulling the latest python:3.7 image for one of our application builds which uses pipenv to install modules we began seeing failures.  The executable `/usr/bin/python3` in the image is symlinked to a 3.9.2 binary.  This causes pipenv to fail in unexpected ways as the Pipfile.lock is expecting 3.7.

```
$ docker run -it --rm python:3.7 /bin/bash
root@c0215d19fb19:/var/tmp# ls -ld /usr/bin/python3*
lrwxrwxrwx 1 root root       9 Apr  5  2021 /usr/bin/python3 -> python3.9
-rwxr-xr-x 1 root root 5479736 Feb 28  2021 /usr/bin/python3.9
root@c0215d19fb19:/var/tmp# ls -ld /usr/local/bin/python3*
lrwxrwxrwx 1 root root     9 Aug  2 12:09 /usr/local/bin/python3 -> python3.7
lrwxrwxrwx 1 root root    16 Aug  2 12:09 /usr/local/bin/python3-config -> python3.7-config
-rwxr-xr-x 2 root root 18400 Aug  2 12:09 /usr/local/bin/python3.7
lrwxrwxrwx 1 root root    17 Aug  2 12:09 /usr/local/bin/python3.7-config -> python3.7m-config
-rwxr-xr-x 2 root root 18400 Aug  2 12:09 /usr/local/bin/python3.7m
-rwxr-xr-x 1 root root  2894 Aug  2 12:09 /usr/local/bin/python3.7m-config
root@c0215d19fb19:/var/tmp# /usr/bin/python3 --version
Python 3.9.2
root@c0215d19fb19:/var/tmp# /usr/local/bin/python3 --version
Python 3.7.13
```

The previous version of the 3.7 tag pointed to 3.7.12-bullseye which did not contain a 3.9 binary.

```
$ docker run -it --rm python:3.7.12-bullseye /bin/bash
root@d5660924c6eb:/# which python3
/usr/local/bin/python3
root@d5660924c6eb:/# cd /usr/bin
root@d5660924c6eb:/usr/bin# ls -ld python3*
ls: cannot access 'python3*': No such file or directory
root@d5660924c6eb:/usr/bin# python3 --version
Python 3.7.12
```

**Comment by yosifkit** [comment]
`/usr/bin/python3` is from the `python3` package (and has been `3.9.x` in all releases of `bullseye`). There are many system-level packages/tools that use Python that we can't reasonably replace with the python that we add that is compiled from source (`/usr/local/bin/python3`).

The 3.9 binary from Debian packages is there in the `python:3.7.12-bullseye` image (and every `-bullseye` image).
```console
$ docker run -it --rm python:3.7.12-bullseye bash
Unable to find image 'python:3.7.12-bullseye' locally
3.7.12-bullseye: Pulling from library/python
e4d61adff207: Pull complete 
4ff1945c672b: Pull complete 
ff5b10aec998: Pull complete 
12de8c754e45: Pull complete 
ada1762e7602: Pull complete 
2f2b2e030155: Pull complete 
8752dc84acc9: Pull complete 
9bc84992c7bf: Pull complete 
356f512becef: Pull complete 
Digest: sha256:b48983bebd0fe1c09639fa008e4cb51aac6277af6c6762fc58ac9d2cb7fc24ef
Status: Downloaded newer image for python:3.7.12-bullseye
root@50c19f82edfc:/# ls -l /usr/bin/python3
lrwxrwxrwx 1 root root 9 Apr  5  2021 /usr/bin/python3 -> python3.9
```

If you were previously using a `-buster` image, then it might not hit the problem because the `python3` package in Debian Buster is `3.7.x` (though there still might be issues or incompatibilities between the two since they might not be the same exact version).

**Comment by mmraz** [comment]
Until the 3.7.13 versions were released and the `python:3.7` tag migrated to the same hash as 3.7.13-bullseye the failure did not occur.  We'd been successfully using the 3.7 tag for our builds.  I'm reporting the bug as I see it, though the missing /usr/bin/python3.9 differs from what you're showing.

Perhaps this problem is related to an accompanying system library for the newer version as well and the version under /usr/bin is a red herring.  We've changed our tag to 3.7.12-bullseye and the builds are passing again.
